src/OrchardCore/OrchardCore.Data.YesSql/ConnectionValidator.cs

```diff
@@ -0,0 +1,88 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient;
+using Microsoft.Data.Sqlite;
+using MySqlConnector;
+using Npgsql;
+using YesSql;
+using YesSql.Provider.MySql;
+using YesSql.Provider.PostgreSql;
+using YesSql.Provider.Sqlite;
+using YesSql.Provider.SqlServer;
+using YesSql.Services;
+using YesSql.Sql;
+
+namespace OrchardCore.Data;
+
+public class ConnectionValidator : IConnectionValidator
+{
+    public async Task<ConnectionValidatorResult> ValidateAsync(string providerName, string connectionString, string tablePrefix)
+    {
+        ArgumentNullException.ThrowIfNull(providerName);
+        ArgumentNullException.ThrowIfNull(connectionString);
+
+        var factory = GetFactory(providerName, connectionString);
+
+        using var connection = factory.CreateConnection();
+
+        try
+        {
+            await connection.OpenAsync();
+        }
+        catch (Exception)
+        {
+            return ConnectionValidatorResult.InvalidConnection;
+        }
+
+        var tableConvention = new DefaultTableNameConvention();
+        var selectBuilder = GetSqlBuilder(providerName, tablePrefix);
+
+        selectBuilder.Select();
+        selectBuilder.AddSelector("*");
+        selectBuilder.Table(tableConvention.GetDocumentTable());
+        selectBuilder.Take("1");
+
+        try
+        {
+            var selectCommand = connection.CreateCommand();
+            selectCommand.CommandText = selectBuilder.ToSqlString();
+
+            using var result = await selectCommand.ExecuteReaderAsync();
+
+            // at this point the query succeeded and the table exists
+            return ConnectionValidatorResult.ValidDocumentExists;
+        }
+        catch (Exception)
+        {
+            // at this point we know that the document table does not exist;
+
+            return ConnectionValidatorResult.ValidDocumentDoesNotExists;
+        }
+    }
+
+    private IConnectionFactory GetFactory(string providerName, string connectionString)
+    {
+        return providerName.ToLower() switch
+        {
+            "sqlconnection" => new DbConnectionFactory<SqlConnection>(connectionString),
```

Comment:
I dislike the magic strings, probably we could add an enum-like class that contains all the providers

Reply:
Agreed! I changed this to `Enum` and removed the string from multiple places. Good suggestion